test/functional/apps/dashboard_elements/controls/options_list/options_list_dashboard_interaction.ts

```diff
@@ -294,12 +294,17 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
         await common.navigateToApp('settings');
         await settings.clickKibanaIndexPatterns();
         await settings.clickIndexPatternByName('animals-*');
+
+        const startingCount = parseInt(await settings.getFieldsTabCount(), 10);
         await settings.addRuntimeField(
           FIELD_NAME,
           'keyword',
           `emit(doc['sound.keyword'].value.substring(0, 1).toUpperCase())`
         );
         await header.waitUntilLoadingHasFinished();
+        await retry.try(async function () {
```

Comment:
How about moving logic to inside [addRuntimeField](https://github.com/elastic/kibana/blob/main/test/functional/page_objects/settings_page.ts#L715)? That way, if this is causing other flaky tests, we can solve the problem in a single location.

Reply:
There's only one other test that uses the `settings` version of `addRuntimeField` from what I can tell (`.../test/functional/apps/management/_runtime_fields.ts`), so I was just trying to make the most minimal change on the Presentation side. But I think you're right that, in case others start using this more frequently, it's probably worth adding it on their end. 

Will do :+1: